### PR TITLE
fix CHECK TABLE distributed_table crash issue

### PR DESCRIPTION
**Pull request by chenxing-xc**
If we issue CHECK TABLE statement on distributed table, segment fault error will occur. The problem is that shard_info.pool could be NULL and we shouldn't create RmoteBlockInputStream on NULL connections.

### CODE REVIEW

File: dbms/src/Interpreters/ClusterProxy/DescribeStreamFactory.cpp [hunk 1]
```diff
@@ -33,24 +32,29 @@ void DescribeStreamFactory::createForShard(
         const Context & context, const ThrottlerPtr & throttler,
         BlockInputStreams & res)
 {
-    for (const Cluster::Address & local_address : shard_info.local_addresses)
+    if (shard_info.isLocal())
```
Comment (alexey-milovidov) [inline dbms/src/Interpreters/ClusterProxy/DescribeStreamFactory.cpp]: DESCRIBE query should go to all replicas of a shard.
For example, if a shard has one local and one remote replica, the query must go to both.
And if we have more than one local replicas, we must go to all of them.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
.

**Comment by chenxing-xc** [comment]
I see， I update the fix by commit c966bf0b95c63c9d1718dbcad2dbeeafc26b0800. please help review again.

File: dbms/src/Interpreters/ClusterProxy/DescribeStreamFactory.cpp (hunk 1 above)
Comment (chenxing-xc) [reply dbms/src/Interpreters/ClusterProxy/DescribeStreamFactory.cpp]: I see， I update the fix by commit c966bf0b95c63c9d1718dbcad2dbeeafc26b0800. please help review again

**Comment by alexey-milovidov** [comment]
Ok, thanks!

**Pull request merged**